cmd/dist/snapshot.go

```diff
@@ -0,0 +1,325 @@
+	},
+}
+
+var snapshotCommitCommand = cli.Command{
+	Name:        "commit",
+	Usage:       "commit an active snapshot",
+	ArgsUsage:   "<name <key>",
+	Description: `Commit the Active layer with name provided`,
+	Action: func(context *cli.Context) error {
+		var (
+			root = context.GlobalString("root")
+			name = context.Args().First()
+			key  = context.Args().Get(1)
+		)
+
+		root, err := filepath.Abs(root)
+		if err != nil {
+			return err
+		}
+		log.G(contextpkg.Background()).WithField("root", root).Debug("setting up snapshot driver")
+
+		driver, err := overlay.NewSnapshotter(root)
+		if err != nil {
+			return err
+		}
+
+		if name == "" {
+			cli.ShowSubcommandHelp(context)
+			return fmt.Errorf("name required")
+		}
+
+		if key == "" {
+			cli.ShowSubcommandHelp(context)
+			return fmt.Errorf("key required")
+		}
+
+		return driver.Commit(contextpkg.Background(), name, key)
+	},
+}
+
+var snapshotMountCommand = cli.Command{
+	Name:        "mount",
+	Usage:       "mount an active snapshot",
+	ArgsUsage:   "<dest> <key> [<parent>]",
+	Description: `Mount active snapshot to destination folder.`,
+	Action: func(context *cli.Context) error {
+		var (
+			root   = context.GlobalString("root")
+			dst    = context.Args().First()
+			key    = context.Args().Get(1)
+			parent = context.Args().Get(2)
+		)
+
+		root, err := filepath.Abs(root)
+		if err != nil {
+			return err
+		}
+		log.G(contextpkg.Background()).WithField("root", root).Debug("setting up snapshot driver")
+
+		driver, err := overlay.NewSnapshotter(root)
+		if err != nil {
+			return err
+		}
+
+		if dst == "" {
+			cli.ShowSubcommandHelp(context)
+			return fmt.Errorf("dst required")
+		}
+
+		if key == "" {
+			return fmt.Errorf("key required")
+		}
+
+		log.G(contextpkg.Background()).WithField("key", key).WithField("parent", parent).Info("preparing snapshot")
+
+		mounts, err := driver.Mounts(contextpkg.Background(), key)
+		if err != nil {
+			return err
+		}
+
+		log.G(contextpkg.Background()).WithField("dst", dst).Info("mounting prepared snapshot")
+		if err := containerd.MountAll(mounts, dst); err != nil {
+			log.G(contextpkg.Background()).WithError(err).Error("error mounting")
+			return err
+		}
+		return nil
+	},
+}
+
+var snapshotStatCommand = cli.Command{
+	Name:      "stat",
+	Usage:     "Gives info of snapshot layer",
+	ArgsUsage: "<key> ",
+	Action: func(context *cli.Context) error {
+		var (
+			root = context.GlobalString("root")
+			key  = context.Args().First()
+		)
+
+		root, err := filepath.Abs(root)
+		if err != nil {
+			return err
+		}
+		log.G(contextpkg.Background()).WithField("root", root).Debug("setting up snapshot driver")
+
+		driver, err := overlay.NewSnapshotter(root)
+		if err != nil {
+			return err
+		}
+
+		if key == "" {
+			cli.ShowSubcommandHelp(context)
+			return fmt.Errorf("key required")
+		}
+
+		info, err := driver.Stat(contextpkg.Background(), key)
+		if err != nil {
+			return err
+		}
+		var snaptype string
+		if info.Kind == 0 {
```

Comment:
Let's implement `Stringer` on the `snapshot.Kind` type so we don't have to do this.